File: 02-programacion-basica/15-practica-integrada/scala/ejemplos-guiados/MiniRegistroNotas.scala
import scala.io.StdIn.readLine
import scala.util.Try

object MiniRegistroNotas:
  def leerNota(indice: Int): Double =
    var notaValida: Option[Double] = None

    while notaValida.isEmpty do
      val entrada = readLine(s"Nota $indice (0-20): ")
      val candidata = Try(entrada.toDouble).toOption.filter(n => n >= 0 && n <= 20)
      notaValida = candidata

      if notaValida.isEmpty then
        println("Entrada inválida.")

    notaValida.get

  def main(args: Array[String]): Unit =
    val notas = (1 to 3).map(leerNota)
    val promedio = notas.sum / notas.size
    val estado = if promedio >= 11 then "Aprobado" else "Desaprobado"

    println(f"Promedio: $promedio%.2f | $estado")

File: 02-programacion-basica/15-practica-integrada/java/ejemplos-guiados/MiniRegistroNotas.java
import java.util.Scanner;

public class MiniRegistroNotas {
    static double promedio(double[] notas) {
        double suma = 0;
        for (double nota : notas) {
            suma += nota;
        }
        return suma / notas.length;
    }

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            double[] notas = new double[3];

            for (int i = 0; i < notas.length; i++) {
                double nota;
                do {
                    System.out.print("Nota " + (i + 1) + " (0-20): ");
                    nota = scanner.nextDouble();
                } while (nota < 0 || nota > 20);

                notas[i] = nota;
            }

            double prom = promedio(notas);
            String estado = prom >= 11 ? "Aprobado" : "Desaprobado";
            System.out.printf("Promedio: %.2f | %s%n", prom, estado);
        }
    }
}
